src/rdkafka_fetcher.c

```diff
@@ -655,12 +789,21 @@ rd_kafka_fetch_reply_handle(rd_kafka_broker_t *rkb,
                         rkt = rd_kafka_topic_find0(rkb->rkb_rk, &topic);
                 }
 
+                fetch_tags.Topics[i].TopicId = topic_id;
+
                 rd_kafka_buf_read_arraycnt(rkbuf, &PartitionArrayCnt,
                                            RD_KAFKAP_PARTITIONS_MAX);
+                fetch_tags.Topics[i].PartitionCnt = PartitionArrayCnt;
+                fetch_tags.Topics[i].Partitions =
+                    rd_calloc(PartitionArrayCnt,
+                              sizeof(rd_kafkap_Fetch_reply_tags_Partition_t));
+
 
                 for (j = 0; j < PartitionArrayCnt; j++) {
                         if (rd_kafka_fetch_reply_handle_partition(
-                                rkb, &topic, rkt, rkbuf, request, ErrorCode))
+                                rkb, &topic, rkt, rkbuf, request, ErrorCode,
+                                &fetch_tags.Topics[i].Partitions[j],
+                                &leader_tags_present) == -1)
                                 goto err_parse;
                 }
```

Comment:
In case we had some partitions with leader change we increase the `topics_with_leader_change_cnt`
```suggestion
                }
                if (has_fetch_tags &&
                    FetchTags.Topics[i].partitions_with_leader_change_cnt) {
                        FetchTags.topics_with_leader_change_cnt++;
                }
```

Reply:
Updated